### PR TITLE
Calculate span hash for the given range

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Fixes [#3708](https://github.com/SonarSource/sonarlint-visualstudio/issues/3708)

### CODE REVIEW

File: src/IssueViz/Editor/IIssueSpanCalculator.cs [hunk 1]
```diff
@@ -97,21 +89,28 @@ internal IssueSpanCalculator(IChecksumCalculator checksumCalculator)
 
             var start = new SnapshotPoint(currentSnapshot, startPos);
             var end = new SnapshotPoint(currentSnapshot, endPos);
+            var snapshotSpan = new SnapshotSpan(start, end);
+
+            if (IsHashDifferent(range.LineHash, snapshotSpan.GetText()) &&
```
Comment (rita-sonarsource) [inline src/IssueViz/Editor/IIssueSpanCalculator.cs:94]: I'm really not sure how to test it. We're mocking the entire VS line :/

File: src/IssueViz/Editor/IIssueSpanCalculator.cs [hunk 2]
```diff
@@ -97,21 +89,28 @@ internal IssueSpanCalculator(IChecksumCalculator checksumCalculator)
...
 
-            return new SnapshotSpan(start, end);
+            return snapshotSpan;
         }
 
-        private bool IsLineHashDifferent(ITextRange range, ITextSnapshotLine snapshotLine)
+        private bool IsHashDifferent(string issueHash, string documentText)
         {
-            if (string.IsNullOrEmpty(range.LineHash))
+            if (string.IsNullOrEmpty(issueHash))
```
Comment (georgii-borovinskikh-sonarsource) [inline src/IssueViz/Editor/IIssueSpanCalculator.cs:106]: this check gets called 2 times + looks kind of out of place => remove it out of this method and place it before the main condition `if (IsHashDifferent(....`

File: src/IssueViz/Editor/IIssueSpanCalculator.cs [hunk 3]
```diff
@@ -97,21 +89,28 @@ internal IssueSpanCalculator(IChecksumCalculator checksumCalculator)
...
             {
                 return false;
             }
 
-            var textInSnapshot = snapshotLine.GetText();
-            var snapshotLineHash = checksumCalculator.Calculate(textInSnapshot);
+            var documentTextHash = checksumCalculator.Calculate(documentText);
 
-            return snapshotLineHash != range.LineHash;
+            return documentTextHash != issueHash;
```
Comment (georgii-borovinskikh-sonarsource) [inline src/IssueViz/Editor/IIssueSpanCalculator.cs:113]: Just a question: why is line hash a string and not a number? Looks so weird

File: src/IssueViz/Editor/IIssueSpanCalculator.cs (hunk 1 above)
Comment (georgii-borovinskikh-sonarsource) [reply src/IssueViz/Editor/IIssueSpanCalculator.cs:94]: WDYM by mocking a vs line? What does "line" mean?

Comment (duncanp-sonar) [reply src/IssueViz/Editor/IIssueSpanCalculator.cs:94]: "line" = `Microsoft.VisualStudio.Text.ITextSnapshotLine`

File: src/IssueViz/Editor/IIssueSpanCalculator.cs (hunk 3 above)
Comment (duncanp-sonar) [reply src/IssueViz/Editor/IIssueSpanCalculator.cs:113]: No particular reason. The SQ APIs return the hashes as strings and we didn't change the format when we deserialized them.

**Review by duncanp-sonar — APPROVED** [review]

**Comment by duncanp-sonar** [comment]
Replaced by #3732

**Pull request closed without merging**